### PR TITLE
Fix #226: Parse quoted blocks with spaces (`' {`)

### LINKED ISSUE #226: Quoted blocks with spaces fail to parse `' {`
Quoted code with spaces between the `'` symbol and the `{` open brace, while being parsed correctly by the Scala compiler, are not parsed by vscode-scala-syntax, creating a all red coloring in VSCode and Github:
example

### PR DESCRIPTION
**Pull request by lolgab**
Fixes the linked issue.

### CODE REVIEW

File: tests/unit/quoted.test.scala [hunk 1]
```diff
@@ -5,11 +5,21 @@
 //    ^ constant.numeric.scala
 //      ^ punctuation.section.block.end.scala
 
+   ' 	{ 2 }
```
Comment (lolgab) [inline tests/unit/quoted.test.scala:8]: Here I added both a tab and a space to cover the handling of tab characters I have in the code.

**Review by tgodzik — APPROVED** [review]
Looks good! We had the same issue recently in Scalameta parser.

**Comment by nicolasstucki** [comment]
That does not seem right. I believe that the syntax requires the quote to be immediately followed by a curly brace. I will double check.

**Comment by tgodzik** [comment]
> That does not seem right. I believe that the syntax requires the quote to be immediately followed by a curly brace. I will double check.

It doesn't actually. At least last time I checked it did compile with whitespaces :O

**Comment by nicolasstucki** [comment]
Indeed. I opened an issue to fix this in https://github.com/lampepfl/dotty/issues/13951.

For 
```scala 
' 	{ 2 }
' [ String ]
```
it looks like the current highlighting is correct for both Scala 2 and Scala 3 (after bug fix) as it is an unclosed `Char` or a `Char` with illegal contents.

**Comment by lolgab** [comment]
Thank you @nicolasstucki for fixing the underlying issue :)

**Comment by tgodzik** [comment]
@nicolasstucki  should we reopen this PR since it turns out it's a correct behaviour?

**Comment by lolgab** [comment]
I'm reopening since the compiler is not changing behaviour.

**Comment by nicolasstucki** [comment]
This is still missing some cases, see and other comments in that thread https://github.com/scala/vscode-scala-syntax/pull/227#discussion_r777565313

**Comment by nicolasstucki** [comment]
This was fixed in #250

**Pull request closed without merging**